nilearn/plotting/img_plotting.py

```diff
@@ -2409,3 +2412,251 @@ def plot_img_comparison(
+
+
+@fill_doc
+def plot_bland_altman(
+    ref_img,
+    src_img,
+    masker=None,
+    ref_label="reference image",
+    src_label="source image",
+    figure=None,
+    title=None,
+    cmap="inferno",
+    gridsize=100,
+    lims=None,
+    output_file=None,
+):
+    """Create a Bland-Altman plot between 2 images.
+
+    Plot the the 2D dsitribution of voxel-wise differences
+    as a function of the voxel-wise mean.
+
+    .. note::
+
+        Bland-Altman plots show
+        the difference between the statistic values (y-axis)
+        against the mean statistic value (x-axis) for all voxels.
+        The plots provide an assessment of the level of agreement
+        between two images about the magnitude of the statistic value
+        observed at each voxel.
+        If two images were in perfect agreement,
+        all points on the bland-altman plot would lie on the x-axis,
+        since the difference between the statistic values
+        at each voxel would be zero.
+        The degree of disagreement is therefore evaluated
+        by the perpendicular distance of points from the x-axis.
+
+    Parameters
+    ----------
+    ref_img : 3D nifti_like
+        Reference image.
+
+    src_img : 3D nifti_like
+        Source image.
+        If the source image is Niimg-Like,
+        it will be resampled to match that or the source image.
+
+    masker : Nifti_like to use as mask or NiftiMasker object or None
+        Mask to be used on data.
+        If None is passed,
+        a NiftiMasker will be fitted on the reference image.
+
+    ref_label : :obj:`str`, default='reference image'
+        Name of reference image.
+
+    src_label : :obj:`str`, default='source image'
+        Name of source image.
+
+    %(figure)s
+
+    %(title)s
+
+    %(cmap)s
+        Default=`inferno`.
+
+    gridsize : :obj:`int` or :obj:`tuple` of 2 :obj:`int`, default=100
+        Dimension of the grid on which to display the main plot.
+        If a single value is passed, then the grid is square.
+        If a tuple is passed, the first value corresponds
+        to the length of the x axis,
+        and the second value corresponds to the length of the y axis.
+
+    %(output_file)s
+
+    Notes
+    -----
+    This function and the plot description was adapted
+    from :footcite:t:`Bowring2019`
+    and its associated `code base <https://github.com/AlexBowring/Software_Comparison/blob/master/figures/lib/bland_altman.py>`_.
+
+
+    References
+    ----------
+
+    .. footbibliography::
+
+
+    """
+    mean, diff = _bland_altman_values(ref_img, src_img, masker=masker)
+
+    if lims is None:
+        lim_x = np.max(np.abs(mean))
+        if lim_x == 0:
+            lim_x = 1
+        lim_y = np.max(np.abs(diff))
+        if lim_y == 0:
+            lim_y = 1
+        lims = [-lim_x, lim_x, -lim_y, lim_y]
+
+    if isinstance(gridsize, int):
+        gridsize = (gridsize, gridsize)
+
+    if figure is None:
+        figsize = (15, 15)
```

Comment:
Do we really need such a huge figsize ? These tends to dwarf the labels, legends, titles...

Reply:
probably not: I still need to tweak things for the defaults

also for now I may limit the number of things that can be tweaked via the API of the function